### ISSUE TITLE
Duplicate Metric Value Forms

### ISSUE THREAD

**Issue by philipbaileynar** [Riverscapes/QRiS#203]
Fix the issue that was occuring during the demo to Joe this morning you were getting multiple metric value forms popping up.

I have a sneaking suspicion that when you investigate this issue you will find that the code looks good and appears to have no problems. I suspect that this has something to do with multiple "ghost" instances of the demo running! If you can't find a bug in the code, this this:

1. Stop QGIS
2. Start QGIS.
3. Go straight to analysis and try editing a metric.
4. Does the form appear only once?
5. Click the plugin reload button.
6. Repeat steps 3 and 4

In other words, I am still convinced that we have not totally solved the lifecycle of our plugin correctly. 

This might explain why I am getting multiple forms pop up when I right click in RAVE to "import to QRiS".

**Comment by KellyMWhitehead** [comment]
My theory is that we are adding signals to the analysis dockable window each time we reopen it. Similar issue to qrave integration adding multiple "Add to QRiS" menu items.

Added some code to 
1) explicitly close the analysis dock window when visibility changes (i.e. the user clicks the X button)
2) explicitly destroy connected signals when the analysis dock widget close event is fired. 

I'm not sure if #2 is overkill, but probably doesn't hurt to clean up dangling signals (probably should do this in other places as well?)

This seems to have solved mulitiple edit windows opening for me, but can revisit if this crops up again

**Comment by philipbaileynar** [comment]
That makes total sense Kelly!

Another idea along these lines is to check the list of attached signals before adding a new one. i.e. if you find the relevant item is already attached to the signal then don't add a new one.

**Comment by KellyMWhitehead** [comment]
FYI use the recievers method on Qtwidgets to search for specific signals that are hooked up
```
if self.table.receivers(self.table.doubleClicked) > 0:
    self.table.doubleClicked.disconnect()
```

**Comment by KellyMWhitehead** [comment]
I haven't seen the duplicate forms since I've made this change. Reopen if this issue pops up again.
